Lib/test/_test_multiprocessing.py

```diff
@@ -2611,6 +2620,41 @@ def errback(exc):
         p.close()
         p.join()
 
+    def test_broken_process_pool1(self):
+        from multiprocessing.pool import BrokenProcessPool
+        p = multiprocessing.Pool(2)
+        res = p.map_async(waiting, range(10))
```

Comment:
You should also test `apply_async` and `imap_unordered`?

Reply:
Added two more tests, one for each